api/src/main/java/org/eclipse/microprofile/faulttolerance/Retry.java

```diff
@@ -101,17 +110,19 @@
 
 
     /**
+     * The list of exception types which should trigger a retry.
      *
-     * @return Specify the failure to retry on. Only a type assignable from {@link java.lang.Error} or {@link java.lang.Exception} can be specified.
-     * Specifying a custom java.lang.Throwable will be ignored.
```

Comment:
Why did you delete this sentence?

Reply:
I sensed the reason when I saw your comments on the PR from @Azquelt. I advise to put the info back on as many people told me that they don't read spec and rely on JavaDoc. For some important info, we need to put on JavaDoc, which is why JDK classes have long javadocs. In Config, I have PR to add some important info from spec to JavaDoc.

Reply:
I propose to _allow_ specifying `Throwable`.

Reply:
What do you mean @Ladicek ! I prefer the sentences are not deleted.

Reply:
What I mean is that it doesn't make any sense to ignore `@Retry(retryOn = Throwable.class)`.

Reply:
I agree with @Ladicek, the current doc specifies that `@Retry(retryOn = Throwable.class)` will be ignored, which is very unintuitive.

We already specify that if a method throws an object which isn't a subclass of `Error` or `Exception` then the behaviour is undefined, we don't also need to have a restriction here that states it must be _ignored_.

I think that `@Retry(retryOn = Throwable.class` should retry on `Exception`, `Error` and all their subclasses. If a throwable that doesn't extend `Exception` or `Error` is thrown, then non-portable behaviour results (as specified elsewhere).

Reply:
@Ladicek I talked with @Azquelt . The bit mentioned in his last paragraph needs to be in the JavaDoc. The current Javadoc does not convene that info, as I think this info is very important.

Reply:
Updated.

I actually consider this a very rare corner case that _doesn't_ warrant 3 mentions in a javadoc of this annotation, but so be it.